src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs

```diff
@@ -102,6 +103,21 @@
            (create-field name (display-value value)))
          log-map)])))
 
+(defn- failure-tree [failures]
+  [comps/Tree {:start-collapsed? false
+               :label "Failures"
+               :highlight-ends false
+               :data (prewalk
+                       (fn [elem]
+                         (cond
+                           ;; reorder map elements to message then causedBy
+                           (map? elem) { "message" (elem "message")
```

Comment:
I could remove the empty `causedBy[]`s here too, but I didn't see an elegant way to do it, and it felt natural to do it on the next step inward.